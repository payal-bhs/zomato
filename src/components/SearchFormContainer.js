import React, { useState } from "react";
import "./SearchForm.style.scss";
import { useDispatch, useSelector } from "react-redux";
import { fetchRestaurant } from "../store/restaurant/restaurantActions";

export const SearchFormContainer = () => {
    const stateSearch = useSelector(state => state.search);
    const dispatch = useDispatch();

    const [city, setCity] = useState(stateSearch.city);
    const [refine, setRefine] = useState(stateSearch.refine);

    const handleSubmit = event => {
        event.preventDefault();
        dispatch(fetchRestaurant({city, refine}));
    };

  return (
    <section title="form" className="search">
        <form
            action=""
            method="post"
            className="search-form"
            id="searchForm"
            name="searchForm"
            onSubmit={handleSubmit}
        >
            <section data-name="city">
                <label className="control-label" htmlFor="city">City</label>
                <input
                    className="form-control"
                    id="city"
                    name="city"
                    type="text"
                    required={true}
                    minLength={3}
                    placeholder="Search for city.."
                    onChange={e => setCity(e.target.value)}
                    value={city}
                />
            </section>
            <section data-name="refine">
                <label className="control-label" htmlFor="refine">Refine</label>
                <input
                    className="form-control"
                    id="refine"
                    name="refine"
                    type="text"
                    placeholder="Refine Search.."
                    onChange={e => setRefine(e.target.value)}
                    value={refine}
                />
            </section>
            <section data-name="actions" className="full-width">
                {/* <button type="submit" className="button" name="submit" id="submit" aria-label="submit search form" onSubmit={handleSubmit}> Search </button> */}
                <input type="submit" value="Submit" id="submit" name="submit" aria-label="submit search form" onSubmit={handleSubmit} />
            </section>
        </form>
    </section>
  );
};

export default SearchFormContainer;

