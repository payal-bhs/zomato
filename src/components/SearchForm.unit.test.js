import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { SearchFormContainer } from "./SearchFormContainer"

// TODO : move this store to test utils
const initialState = {
    search: {
        city: "",
        refine: ""
    },
    restaurant : {
        loading: false,
        restaurants: [],
        error: ""
    }
};

const mockStore = configureStore();
const defaultStore = mockStore(initialState);


describe("Search Form", () => {

    it("render filter layout for restaurant", () => {
        const restaurantFilter = shallow(
            <Provider store={defaultStore}>
                <SearchFormContainer />
            </Provider>);
        expect(restaurantFilter).toMatchSnapshot();
    });
});
