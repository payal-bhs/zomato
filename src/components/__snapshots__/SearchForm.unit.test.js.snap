// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form render filter layout for restaurant 1`] = `ShallowWrapper {}`;
